src/main/java/ch/njol/skript/expressions/ExprSpawner.java

```diff
@@ -0,0 +1,131 @@
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.entity.EntityType;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Spawner Type")
+@Description("Retrieves, sets, or resets the spawner's entity type")
+@Examples({"broadcast \"%spawner's entity type%\""})
+@Since("INSERT VERSION")
+public class ExprSpawner extends SimplePropertyExpression<Block, EntityData> {
+	
+	private static final Material MATERIAL_SPAWNER = Aliases.javaItemType("spawner").getMaterial();
+	private static final Map<EntityType, org.bukkit.entity.EntityType> CACHE = new HashMap<EntityType, org.bukkit.entity.EntityType>();
+	
+	static {
+		//Cache Bukkit EntityType -> Skript EntityType 
+		for(org.bukkit.entity.EntityType e : org.bukkit.entity.EntityType.values()) {
+			//Replace underscores with spaces to comply with Skript's Alias format 
+			CACHE.put(EntityType.parse(e.toString().replaceAll("_", " ")), e);
+		}
+		register(ExprSpawner.class, EntityData.class, "entity type", "blocks");
+	}
+	
+	@Override
+	@Nullable
+	public EntityData convert(final Block b) {
+		if (b.getType() != MATERIAL_SPAWNER)
+			return null;
+		return toSkriptEntityData(((CreatureSpawner)b.getState()).getSpawnedType());
+	}
+	
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(Changer.ChangeMode mode) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.RESET) 
+			return CollectionUtils.array(EntityType.class);
+		return null;
+	}
+	
+	@SuppressWarnings("null")
+	@Override
+	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
+		for (Block b : getExpr().getArray(e)) {
+			if (b.getType() != MATERIAL_SPAWNER)
+				continue;
+			CreatureSpawner s = (CreatureSpawner) b.getState();
+			switch (mode) {
+				case SET:
+					s.setSpawnedType(toBukkitEntityType((EntityType) delta[0]));
+					break;
+				case RESET:
+					s.setSpawnedType(org.bukkit.entity.EntityType.PIG);
+					break;
+			}
+			s.update(); //Actually trigger the spawner's update 
+		}
+	}
+	
+	@Override
+	public Class<EntityData> getReturnType() {
+		return EntityData.class;
+	}
+	
+	@Override
+	protected String getPropertyName() {
+		return "entity type";
+	}
+	
+	/**
+	 * Convert from Skript's EntityType to Bukkit's EntityType
+	 * @param e Skript's EntityType
+	 * @return Bukkit's EntityType
+	 */
+	@SuppressWarnings({"null"})
+	public static org.bukkit.entity.EntityType toBukkitEntityType(EntityType e){
+		return CACHE.get(e);
+	}
+	
+	/**
+	 * Convert from Bukkit's EntityType to Skript's EntityData
+	 * @param e Bukkit's EntityType
+	 * @return Skript's EntityData
+	 */
+	@SuppressWarnings("null")
+	public static EntityData toSkriptEntityData(org.bukkit.entity.EntityType e){
+		//Replace underscores with spaces to comply with Skript's Alias format 
+		return EntityData.parse(e.toString().replaceAll("_", " "));
```

Comment:
So again unsafe text parsing, can use `new EntityType(entity class)` but this also works with a loop, i guess you should make another map for that? 😖

Reply:
Added a 2nd cache map for Bukkit EntityType -> Skript EntityData